src/lib.rs

```diff
@@ -91,6 +91,8 @@ pub struct IpfsOptions<Types: IpfsTypes> {
     pub bootstrap: Vec<(Multiaddr, PeerId)>,
     /// Enables mdns for peer discovery when true.
     pub mdns: bool,
+    /// Custom Kademlia protocol name.
+    pub kad_protocol: Option<String>,
```

Comment:
could this be left out for now at least? Probably later on there should be custom config for which protocol ids to enable, but probably not now (to avoid "/lan" and !"/lan" confusion found in the PR already)

Reply:
perhaps I can remove the values in the calls, but leave the option itself? seems innocuous and we're likely to want it later on

Reply:
Yeah definitely no need give this in the inmemory_with_generated_keys since it's not needed in the existing tests at least.

Could you elaborate why we'd like to have this later on?

Reply:
so we can test both protocols; that recent go-ipfs article suggested they are both significant

Reply:
> recent go-ipfs article suggested they are both significant

I think that is for multiple DHTs, not either one of those: lan for private networks and the !lan for the global. I think we should be fine with just the global if anything at the moment.
From above:

>> Yeah definitely no need give this in the inmemory_with_generated_keys since it's not needed in the existing tests at least.

Do you agree we should delete the parameter from `inmemory_with_generated_keys`? I see no value in configuring it in all test scenarios, maybe in some of the dht specific, or if we want to use something specific which is sure to be internal for our test case.

Reply:
>Do you agree we should delete the parameter from inmemory_with_generated_keys?

yeah, I just meant leaving it in `IpfsOptions`